### PR TITLE
feat: add auth middleware

### PR DESCRIPTION
**Pull request by n-r-b**
### JIRA
https://companieshouse.atlassian.net/browse/SR-82

### Change Description

- Injected authMiddleware from `web-securty-node` lib into main app
- Auth middleware redirects user to sign in screen and then returns them to the applicant details page
- Landing page is excluded from sign in
- Updated session structure in `SessionService` and updated tests to reflect this
- Injected session/auth middleware mock into all controller tests (next PR will contain auth tests covering all routes)

### Page Screenshot
N/A

### Work Checklist

- [x] Tests added where applicable
- [x] Test coverage of new code meets target of 80%
- [x] UI changes look good on mobile
- [x] UI changes meet accessibility criteria

---

### Merge Instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: test/services/SessionService.test.ts [hunk 1]
```diff
@@ -1,70 +1,75 @@
...
+
+const mockRequestData = {
+  session: {
+    get: jest.fn(),
+    data: jest.fn(),
+    getExtraData: jest.fn(),
+    setExtraData: jest.fn(),
+    deleteExtraData: jest.fn(),
+    verify: jest.fn()
+  } as unknown as Session
```
Comment (geek-soft) [inline test/services/SessionService.test.ts:34]: What `as unknown as Session` does? Does it set type to `Session` or `Session | undefined`?

**Review by akaJSaunders — APPROVED** [review]

File: src/services/SessionService.ts [hunk 2]
```diff
@@ -1,14 +1,13 @@
 import { Request } from 'express';
 import { SuppressionData, SUPPRESSION_DATA_KEY } from '../models/SuppressionDataModel';
 
-
 export default class SessionService {
 
   static getSuppressionSession(req: Request): SuppressionData | undefined {
```
Comment (geek-soft) [inline src/services/SessionService.ts:6]: Passing request while only all operations are performed on session instance does not look good. Possible improvement in the future. Not this PR though.

File: test/services/SessionService.test.ts [hunk 3]
```diff
@@ -1,70 +1,75 @@
...
+    companyName: 'company-name-test',
+    companyNumber: 'NI000000',
+    description: 'This is a document',
+    date: '2020-01-01'
+  }
+};
+
+const mockRequestData = {
+  session: {
+    get: jest.fn(),
```
Comment (geek-soft) [inline test/services/SessionService.test.ts:28]: Mocking whole session object while only one method `getExtraData` is needed seems unnecessary.

File: test/services/SessionService.test.ts [hunk 4]
```diff
@@ -1,70 +1,75 @@
...
-        date: '2020-01-01'
-      }
-    };
-
-    const mockRequest: any = {
-      session: { data: { extra_data: { suppression:  suppressionData  } } }
-    };
-
-    expect(SessionService.getSuppressionSession(mockRequest)).toEqual(suppressionData)
+    const suppressionData: SuppressionData = mockSuppressionData;
```
Comment (geek-soft) [inline test/services/SessionService.test.ts:41]: Why `mockSuppressionData` needs to be reassigned to `suppressionData`? It is like:

```
const a = 1
const b = a
b...
```

I see no value.

**Review by geek-soft — COMMENTED** [review]
Changes look good although SessionService test could be refactored to make it easier to follow.

File: test/services/SessionService.test.ts (hunk 1 above)
Comment (n-r-b) [reply test/services/SessionService.test.ts]: managed to get this to work with setting the type to `Session`, but `as any` is needed for the `getExtraData` type

File: test/services/SessionService.test.ts (hunk 3 above)
Comment (n-r-b) [reply test/services/SessionService.test.ts]: removed other properties

File: test/services/SessionService.test.ts (hunk 4 above)
Comment (n-r-b) [reply test/services/SessionService.test.ts]: tidied this up

**Review by geek-soft — APPROVED** [review]

**Pull request merged**